src/LuaTransformer.ts

```diff
@@ -3589,7 +3620,11 @@ export class LuaTransformer {
                 } else {
                     const arg1 = params[0];
                     const arg2 = params[1];
-                    const sumArg = tstl.createBinaryExpression(arg1, arg2, tstl.SyntaxKind.AdditionOperator);
+                    const sumArg = tstl.createBinaryExpression(
+                        tstl.createParenthesizedExpression(arg1),
```

Comment:
This seems like a random parenthesization?

Reply:
if the arg is a complex expression (such as another binary), it needs parenthesis to avoid order-of-operation issues. Now that I re-look at this, the second one should probably be parenthesized as well.